fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/DataCachePopulateRewriteRule.java

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.rule.tree;
+
+import com.starrocks.catalog.Table;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.datacache.DataCacheOptions;
+import com.starrocks.datacache.DataCachePopulateMode;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.SessionVariable;
+import com.starrocks.sql.ast.QueryStatement;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptExpressionVisitor;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.ScanOperatorPredicates;
+import com.starrocks.sql.optimizer.operator.physical.PhysicalScanOperator;
+import com.starrocks.sql.optimizer.task.TaskContext;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+public class DataCachePopulateRewriteRule implements TreeRewriteRule {
+    private static final Logger LOG = LogManager.getLogger(DataCachePopulateRewriteRule.class);
+
+    private final ConnectContext connectContext;
+
+    public DataCachePopulateRewriteRule(ConnectContext connectContext) {
+        this.connectContext = connectContext;
+    }
+
+    @Override
+    public OptExpression rewrite(OptExpression root, TaskContext taskContext) {
+        SessionVariable sessionVariable = connectContext.getSessionVariable();
+        if (!sessionVariable.isEnableScanDataCache()) {
+            return root;
+        }
+        DataCachePopulateMode populateMode = getPopulateMode(sessionVariable);
```

Comment:
early return if it's NEVER

Reply:
Actually, we can't return `NEVER` early because of compatible reasons.

If we return directly, it means we will not set `DataCacheOptions` in ScanNode. Then in `be`, it will set populate behavior by the `enable_populate_datacache` session variable, whose default value is true.